src/chat-api/store/useChannels.ts

```
@@ -120,9 +130,47 @@ function hasPermission(
   return hasBit(permissions, bitwise.bit);
 }
 
+function canSendMessage(
+  this: Channel,
+  userId: string
+) {
+  const serverMembers = useServerMembers();
+  const account = useAccount();
+
+  const member = this.serverId ? serverMembers.get(this.serverId, userId) : undefined;
+  const muted = member?.muteExpireAt && new Date(member?.muteExpireAt!) > new Date();
```

Biome (2.3.14)

[error] Forbidden non-null assertion after optional chaining.

Optional chaining already handles nullish values. Using non-null assertion defeats its purpose and may cause runtime errors.
Consider using the nullish coalescing operator `??` or optional chaining throughout the chain instead.

(lint/suspicious/noNonNullAssertedOptionalChain)